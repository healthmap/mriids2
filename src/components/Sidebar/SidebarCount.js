import React from "react";
import { connect } from "react-redux";
import { BlockPadded } from "../styled-components/Block";
import {
  SidebarCountParent,
  SidebarCountLabel,
  SidebarCountValue,
} from "../styled-components/SidebarCountStyles";
import {
  getEbolaCaseCount,
  getFutureProjectionCount,
} from "../../utils/ebolaDataHelpers";
import { getCovidCount } from "../../utils/covidDataHelpers";

const SidebarCount = ({
  filters,
  ebolaData,
  ebolaDataCombined,
  covidCaseCountData,
  covidDeathCountData,
}) => {
  // Gets the reported case count for either the Ebola or COVID 19 outbreak.
  const reportedCaseCount =
    filters.outbreak === "COVID 19"
      ? getCovidCount(covidCaseCountData, filters)
      : getEbolaCaseCount(ebolaData, filters);

  // Gets the reported death count for the COVID 19 outbreak.
  const reportedDeathCount = getCovidCount(covidDeathCountData, filters);

  // Gets the projected case count for the ebola outbreak.
  const projectedDiseaseCount = getFutureProjectionCount(
    ebolaData,
    ebolaDataCombined,
    filters
  );

  const dataType =
    filters.outbreak === "COVID 19" ? "cases and deaths" : "cases";
  const locationText =
    filters.country === "All" ? "all locations" : filters.country;
  const titleText = filters.dataType.includes("projected")
    ? `Projection in ${locationText}`
    : `Reported ${dataType} in ${locationText}`;

  return (
    <BlockPadded>
      <p>
        {titleText} from:
        <br />
        <strong>
          {filters.dateRange.from.toDateString()} {" - "}
          {filters.dateRange.to.toDateString()}
        </strong>
      </p>
      <SidebarCountParent>
        <SidebarCountLabel>REPORTED CASES</SidebarCountLabel>
        <SidebarCountValue>
          {reportedCaseCount.toLocaleString()}
        </SidebarCountValue>
      </SidebarCountParent>
      {filters.outbreak === "COVID 19" && (
        <SidebarCountParent>
          <SidebarCountLabel>REPORTED DEATHS</SidebarCountLabel>
          <SidebarCountValue>
            {reportedDeathCount.toLocaleString()}
          </SidebarCountValue>
        </SidebarCountParent>
      )}
      {filters.dataType.includes("projected") && (
        <SidebarCountParent>
          <SidebarCountLabel>Projected future cases</SidebarCountLabel>
          <SidebarCountValue>
            {projectedDiseaseCount.toLocaleString()}
          </SidebarCountValue>
        </SidebarCountParent>
      )}
    </BlockPadded>
  );
};

const mapStateToProps = (state) => ({
  filters: state.filters,
  ebolaData: state.ebola.ebolaData.data,
  ebolaDataCombined: state.ebola.ebolaDataCombined.data,
  covidCaseCountData: state.covid.caseCounts.data,
  covidDeathCountData: state.covid.deathCounts.data,
});

export default connect(mapStateToProps)(SidebarCount);
