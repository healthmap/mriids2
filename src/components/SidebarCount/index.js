import React from "react";
import { connect } from "react-redux";
import {
  getEbolaCaseCount,
  getFutureProjectionCount,
} from "../../utils/ebolaDataHelpers";
import { getCovidCount } from "../../utils/covidDataHelpers";
import { BlockPadded } from "../SharedStyledComponents/Block";
import * as Styled from "./styles";

const SidebarCount = ({
  filters,
  ebolaData,
  ebolaDataCombined,
  covidCaseCountData,
  covidDeathCountData,
}) => {
  // Gets the reported case count for either the Ebola or COVID 19 outbreak.
  const reportedCaseCount =
    filters.outbreak === "COVID 19"
      ? getCovidCount(covidCaseCountData, filters)
      : getEbolaCaseCount(ebolaData, filters);

  // Gets the reported death count for the COVID 19 outbreak.
  const reportedDeathCount = getCovidCount(covidDeathCountData, filters);

  // Gets the projected case count for the ebola outbreak.
  const projectedDiseaseCount = getFutureProjectionCount(
    ebolaData,
    ebolaDataCombined,
    filters
  );

  const dataType =
    filters.outbreak === "COVID 19" ? "cases and deaths" : "cases";
  const locationText =
    filters.country === "All" ? "All Locations" : filters.country;
  const titleText =
    filters.dataType === "projected cases"
      ? "Projection in"
      : `Total reported ${dataType} in`;
  const reportedCasesSectionTitle =
    filters.dataType === "projected cases"
      ? "Total outbreak projections"
      : "Reported cases";

  return (
    <BlockPadded>
      <p>
        {titleText} <strong>{locationText}</strong> from{" "}
        <strong>
          {filters.dateRange.from.toDateString()}
          {"â€“"}
          {filters.dateRange.to.toDateString()}
        </strong>
      </p>
      <Styled.SidebarCountParent>
        <Styled.SidebarCountLabel>
          {reportedCasesSectionTitle}
        </Styled.SidebarCountLabel>
        <Styled.SidebarCountValue>
          {reportedCaseCount.toLocaleString()}
        </Styled.SidebarCountValue>
      </Styled.SidebarCountParent>
      {filters.outbreak === "COVID 19" && (
        <Styled.SidebarCountParent>
          <Styled.SidebarCountLabel>Reported Deaths</Styled.SidebarCountLabel>
          <Styled.SidebarCountValue>
            {reportedDeathCount.toLocaleString()}
          </Styled.SidebarCountValue>
        </Styled.SidebarCountParent>
      )}
      {filters.dataType === "projected cases" && (
        <Styled.SidebarCountParent>
          <Styled.SidebarCountLabel>
            Projected future cases
          </Styled.SidebarCountLabel>
          <Styled.SidebarCountValue>
            {projectedDiseaseCount.toLocaleString()}
          </Styled.SidebarCountValue>
        </Styled.SidebarCountParent>
      )}
    </BlockPadded>
  );
};

const mapStateToProps = (state) => ({
  filters: state.filters,
  ebolaData: state.ebola.ebolaData.data,
  ebolaDataCombined: state.ebola.ebolaDataCombined.data,
  covidCaseCountData: state.covid.caseCounts.data,
  covidDeathCountData: state.covid.deathCounts.data,
});

export default connect(mapStateToProps)(SidebarCount);
