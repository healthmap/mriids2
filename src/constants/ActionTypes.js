export const FETCH_EBOLA_DATA_REQUEST = "FETCH_EBOLA_DATA_REQUEST";
export const FETCH_EBOLA_DATA_SUCCESS = "FETCH_EBOLA_DATA_SUCCESS";
export const FETCH_EBOLA_DATA_FAILURE = "FETCH_EBOLA_DATA_FAILURE";

export const FETCH_EBOLA_DATA_COMBINED_REQUEST =
  "FETCH_EBOLA_DATA_COMBINED_REQUEST";
export const FETCH_EBOLA_DATA_COMBINED_SUCCESS =
  "FETCH_EBOLA_DATA_COMBINED_SUCCESS";
export const FETCH_EBOLA_DATA_COMBINED_FAILURE =
  "FETCH_EBOLA_DATA_COMBINED_FAILURE";

export const FETCH_RISK_DATA_REQUEST = "FETCH_RISK_DATA_REQUEST";
export const FETCH_RISK_DATA_SUCCESS = "FETCH_RISK_DATA_SUCCESS";
export const FETCH_RISK_DATA_FAILURE = "FETCH_RISK_DATA_FAILURE";

export const FETCH_COVID_CASE_COUNT_DATA_REQUEST =
  "FETCH_COVID_CASE_COUNT_DATA_REQUEST";
export const FETCH_COVID_CASE_COUNT_DATA_SUCCESS =
  "FETCH_COVID_CASE_COUNT_DATA_SUCCESS";
export const FETCH_COVID_CASE_COUNT_DATA_FAILURE =
  "FETCH_COVID_CASE_COUNT_DATA_FAILURE";

export const FETCH_COVID_DATA_COMBINED_REQUEST =
  "FETCH_COVID_DATA_COMBINED_REQUEST";
export const FETCH_COVID_DATA_COMBINED_SUCCESS =
  "FETCH_COVID_DATA_COMBINED_SUCCESS";
export const FETCH_COVID_DATA_COMBINED_FAILURE =
  "FETCH_COVID_DATA_COMBINED_FAILURE";

// These are used for updating the filters
export const CHANGE_COUNTRY = "CHANGE_COUNTRY";
export const CHANGE_OUTBREAK = "CHANGE_OUTBREAK";
export const CHANGE_DATE_RANGE = "CHANGE_DATE_RANGE";
export const CHANGE_VIEW = "CHANGE_VIEW";
export const CHANGE_PROJECTION = "CHANGE_PROJECTION";
