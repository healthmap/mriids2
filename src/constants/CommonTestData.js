export const reduxInitialState = {
  ebola: {
    ebolaData: {
      isFetching: 0,
      data: {},
      error: {},
    },
    ebolaDataCombined: {
      isFetching: 0,
      data: {},
      error: {},
    },
    riskData: {
      isFetching: 0,
      data: {},
      error: {},
    },
  },
  covid: {
    caseCounts: {
      isFetching: 0,
      data: [],
      error: {},
    },
    deathCounts: {
      isFetching: 0,
      data: [],
      error: {},
    },
  },
  filters: {
    country: "All",
    outbreak: "Ebola Outbreak",
    view: "snapshot",
    projection: false,
    dateRange: { from: new Date(2014, 9, 1), to: new Date(2016, 1, 20) },
  },
};

// Same as the reduxInitialState, but with the 'risk' view in the filters.
export const riskViewState = {
  ...reduxInitialState,
  filters: {
    ...reduxInitialState.filters,
    view: "risk",
  },
};
