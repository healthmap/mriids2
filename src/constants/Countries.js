export const ebolaOutbreakCountries = ["Guinea", "Liberia", "Sierra Leone"];

export const allCountries = [
  "Afghanistan",
  "Albania",
  "Algeria",
  "Andorra",
  "Angola",
  "Antigua and Barbuda",
  "Argentina",
  "Armenia",
  "Aruba",
  "Australia",
  "Austria",
  "Azerbaijan",
  "Bahamas",
  "Bahrain",
  "Bangladesh",
  "Barbados",
  "Belarus",
  "Belgium",
  "Belize",
  "Benin",
  "Bhutan",
  "Bolivia",
  "Bosnia and Herzegovina",
  "Botswana",
  "Brazil",
  "British Virgin Islands",
  "Brunei Darussalam",
  "Bulgaria",
  "Burkina Faso",
  "Burundi",
  "Cambodia",
  "Cameroon",
  "Canada",
  "Cape Verde",
  "Cayman Islands",
  "Central African Republic",
  "Chad",
  "Chile",
  "China",
  "Colombia",
  "Comoros",
  "Congo",
  "Costa Rica",
  "Cote dIvoire",
  "Croatia",
  "Cuba",
  "Cyprus",
  "Czechia",
  "Democratic Republic of the Congo",
  "Denmark",
  "Djibouti",
  "Dominica",
  "Dominican Republic",
  "Ecuador",
  "Egypt",
  "El Salvador",
  "Equatorial Guinea",
  "Eritrea",
  "Estonia",
  "Eswatini",
  "Ethiopia",
  "Fiji",
  "Finland",
  "France",
  "Gabon",
  "Gambia",
  "Georgia",
  "Germany",
  "Ghana",
  "Greece",
  "Greenland",
  "Grenada",
  "Guatemala",
  "Guinea",
  "Guinea Bissau",
  "Guyana",
  "Haiti",
  "Holy See",
  "Honduras",
  "Hungary",
  "Iceland",
  "India",
  "Indonesia",
  "Iran",
  "Iraq",
  "Ireland",
  "Israel",
  "Italy",
  "Jamaica",
  "Japan",
  "Jordan",
  "Kazakhstan",
  "Kenya",
  "Kosovo",
  "Kuwait",
  "Kyrgyzstan",
  "Laos",
  "Latvia",
  "Lebanon",
  "Lesotho",
  "Liberia",
  "Libya",
  "Liechtenstein",
  "Lithuania",
  "Luxembourg",
  "Madagascar",
  "Malawi",
  "Malaysia",
  "Maldives",
  "Mali",
  "Malta",
  "Marshall Islands",
  "Mauritania",
  "Mauritius",
  "Mexico",
  "Moldova",
  "Monaco",
  "Mongolia",
  "Montenegro",
  "Morocco",
  "Mozambique",
  "Myanmar",
  "Namibia",
  "Nepal",
  "Netherlands",
  "New Zealand",
  "Nicaragua",
  "Niger",
  "Nigeria",
  "North Macedonia",
  "Norway",
  "Oman",
  "Pakistan",
  "Panama",
  "Papua New Guinea",
  "Paraguay",
  "Peru",
  "Philippines",
  "Poland",
  "Portugal",
  "Puerto Rico",
  "Qatar",
  "Romania",
  "Russia",
  "Rwanda",
  "Saint Kitts and Nevis",
  "Saint Lucia",
  "Saint Vincent and the Grenadines",
  "Samoa",
  "San Marino",
  "Sao Tome and Principe",
  "Saudi Arabia",
  "Senegal",
  "Serbia",
  "Seychelles",
  "Sierra Leone",
  "Singapore",
  "Slovakia",
  "Slovenia",
  "Solomon Islands",
  "Somalia",
  "South Africa",
  "South Korea",
  "South Sudan",
  "Spain",
  "Sri Lanka",
  "Sudan",
  "Suriname",
  "Sweden",
  "Switzerland",
  "Syria",
  "Taiwan",
  "Tajikistan",
  "Thailand",
  "Timor Leste",
  "Togo",
  "Trinidad and Tobago",
  "Tunisia",
  "Turkey",
  "Uganda",
  "Ukraine",
  "United Arab Emirates",
  "United Kingdom",
  "United Republic of Tanzania",
  "United States of America",
  "Uruguay",
  "Uzbekistan",
  "Vanuatu",
  "Venezuela",
  "Vietnam",
  "Yemen",
  "Zambia",
  "Zimbabwe",
];
