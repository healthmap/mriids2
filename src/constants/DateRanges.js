export const ebolaInitialDateRange = {
  from: new Date(2014, 9, 1),
  to: new Date(2016, 1, 20),
};
