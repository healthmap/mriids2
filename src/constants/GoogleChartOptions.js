export const options = {
  isStacked: true,
  bar: {
    groupWidth: "80%",
  },
  colors: ["#4d73ce", "#f8ae32"],
  hAxis: {
    textStyle: {
      fontSize: 11,
    },
    gridlines: {
      color: "#ececec",
    },
    minorGridlines: {
      color: "#ececec",
    },
  },
  vAxis: {
    textStyle: {
      fontSize: 11,
    },
    gridlines: {
      color: "#ececec",
    },
    minorGridlines: {
      color: "#ececec",
    },
    baselineColor: "#ccc",
  },
  legend: {
    textStyle: {
      fontSize: 11,
    },
  },
  chartArea: {
    left: 80,
    width: "75%",
  },
};
