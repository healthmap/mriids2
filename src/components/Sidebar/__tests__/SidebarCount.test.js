import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import SidebarCount from "../SidebarCount";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { reduxInitialState } from "../../../constants/CommonTestData";

const mockStore = configureStore([thunk]);

describe("Tests for the connected SidebarCount component with reduxInitialState", () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore(reduxInitialState);

    component = renderer.create(
      <Provider store={store}>
        <SidebarCount />
      </Provider>
    );
  });
  test("should render with given state from Redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
