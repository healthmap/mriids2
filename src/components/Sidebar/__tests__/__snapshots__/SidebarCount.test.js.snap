// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the connected SidebarCount component with reduxInitialState should render with given state from Redux store 1`] = `
.c0 {
  margin-top: 2rem;
  padding: 1.3rem;
  font-size: 1.2rem;
  border: 1px solid #ccc;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c1 {
  margin: 0.6rem 0;
}

.c2 {
  margin-bottom: 0.3rem;
  text-transform: uppercase;
  font-weight: 600;
}

.c3 {
  display: inline-block;
  font-size: 1.8em;
  font-weight: 600;
}

<div
  className="c0"
>
  <p>
    Reported cases in all locations
     from:
    <br />
    <strong>
      Wed Oct 01 2014
       
       - 
      Sat Feb 20 2016
    </strong>
  </p>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      REPORTED CASES
    </div>
    <div
      className="c3"
    >
      0
    </div>
  </div>
</div>
`;
